framework/db/ActiveRecordInterface.php

```diff
@@ -294,12 +294,12 @@ public static function findAll($condition);
      *
      * @param array $attributes attribute values (name-value pairs) to be saved for the record.
      * Unlike [[update()]] these are not going to be validated.
-     * @param array $condition the condition that matches the records that should get updated.
+     * @param string|array $condition the condition that matches the records that should get updated.
      * Please refer to [[QueryInterface::where()]] on how to specify this parameter.
      * An empty condition will match all records.
      * @return int the number of rows updated
      */
-    public static function updateAll($attributes, $condition = null);
+    public static function updateAll($attributes, $condition = '');
```

Comment:
this change is incorrect, `ActiveRecordInterface` is independet from SQL DB so the correct value for no condition is `null`.

Reply:
@cebe But in implementation of the interface you have `$condition = ''`, and IDE is raising warning there. Maybe it's better not to have default value in interface at all and just document possible values.

Reply:
> IDE is raising warning there

IDE is wrong, changing default value of parameter works perfectly fine in PHP.

Reply:
@rob006 It works fine, but I see it as a violation of contract between the class and interface. If the default parameter is changed in majority of places what is the purpose of having it in interface in the first place?

Reply:
> If the default parameter is changed in majority of places what is the purpose of having it in interface in the first place?

To define parameter which is optional, as a part of the interface. Default value is a implementation detail and it is not part of the contract.